great_expectations/data_context/store/datasource_store.py

```diff
@@ -123,6 +125,28 @@ def delete_by_name(self, datasource_name: str) -> None:
         )
         self.remove_key(datasource_key)
 
+    def delete(self, datasource_config: DatasourceConfig) -> None:
```

Comment:
We also have `remove_key` if you only have access to the key. This is a step towards using the datasource or datasource config for interaction with the store so that users don't have to worry about the id field.